### PR TITLE
Consume EO's delete by percentage API changes

### PR DESCRIPTION
**Pull request by btaani**
### Description
This PR:

- Updates go.mod to use the latest EO image
- Updates CLO's retention policy API to include delete by disk percentage functionality

/cc @periklis 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-2429

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
-LABEL \
-    com.redhat.component="cluster-logging-operator" \
-    version="v1.1" \
-    name="cluster-logging-operator" \
-    License="Apache-2.0" \
-    io.k8s.display-name="cluster-logging-operator bundle" \
-    io.k8s.description="bundle for the cluster-logging-operator" \
-    summary="This is the bundle for the cluster-logging-operator" \
-    maintainer="AOS Logging <aos-logging@redhat.com>"
+COPY bundle/tests/scorecard /tests/scorecard/
```
Comment (periklis) [inline bundle.Dockerfile:21]: This changes look unintended and the core team should address the bundle freshness in a separate PR. For now remove them for this PR.

File: internal/k8shandler/indexmanagement/index_management.go [hunk 2]
```diff
@@ -78,14 +79,17 @@ func newDefaultPoliciesSpec(spec *logging.RetentionPoliciesSpec) *logging.Retent
...
 		},
 		Infra: &logging.RetentionPolicySpec{
 			MaxAge:                  esapi.TimeUnit("7d"),
 			PruneNamespacesInterval: esapi.TimeUnit("30m"),
+			DiskThresholdPercent:    100,
 		},
 		Audit: &logging.RetentionPolicySpec{
 			MaxAge:                  esapi.TimeUnit("7d"),
 			PruneNamespacesInterval: esapi.TimeUnit("30m"),
+			DiskThresholdPercent:    100,
```
[the remark this answers is not in the document]
Comment (btaani) [reply internal/k8shandler/indexmanagement/index_management.go:92]: The unit test kept failing until I added them

**Review by periklis — APPROVED** [review]
/lgtm

**Pull request merged**